### PR TITLE
[Azure Service Bus] Stop copying ReadonlyMemory from original given that we are using an immutable and readonly memory reference

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs [hunk 1]
```diff
@@ -285,9 +285,7 @@ public static ServiceBusMessage CreateFrom(ServiceBusReceivedMessage message)
             var originalBody = message.Body;
             if (!originalBody.IsEmpty)
             {
-                var clonedBody = new byte[originalBody.Length];
-                Array.Copy(originalBody.ToArray(), clonedBody, originalBody.Length);
-                copiedMessage.Body = clonedBody;
+                copiedMessage.Body = originalBody;
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs:288]: Good catch! I wonder can we get away with just moving the Body set into the object initializer with the rest of the properties and doing away with the IsEmpty check?

Comment (danielmarbach) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs:288]: I think that can work. Can do tomorrow or so

Comment (danielmarbach) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs]: Done

**Review by ShivangiReja — APPROVED** [review]
PR looks good to me. Thanks @danielmarbach!

**Pull request merged**